tests/rptest/tests/configuration_update_test.py

```diff
@@ -142,6 +142,10 @@ def broker_configuration_updated():
             client = PythonLibrdkafka(self.redpanda)
             brokers = client.brokers()
             self.logger.debug(f"brokers metadata: {brokers}")
-            return brokers[2].port == 10091
+            # The number of brokers in real life doesn't change, but
+            # the number in the kafka metadata response varies depending
+            # on which nodes are considered up by health monitor: must
+            # check length of brokers list before indexing node 2.
+            return len(brokers) >= 3 and brokers[2].port == 10091
```

Comment:
are brokers always sorted the same way ?

Reply:
In this particular test, it was deterministic because redpanda.py methodically adds each broker in turn, so they end up in the same index in members_table.

But while we're updating it might as well make the test more robust anyway: I've pushed an update.